docs/design/specs/runtime-async.md

```diff
@@ -59,31 +61,30 @@ Other restrictions are likely to be permanent, including
 * Suspension points may not appear in exception handling blocks.
 * Only four types will be supported as the return type for "runtime-async" methods: `System.Threading.Task`, `System.Threading.ValueTask`, `System.Threading.Task<T>`, and `System.Threading.ValueTask<T>`
 
-All async methods effectively have two entry points, or signatures. The first signature is the one present in method definitions: a `Task` or `ValueTask` returning method. The second signature is a "secondary signature", described in further detail in [I.8.6.1.5 Method signatures].
+In addition to the `AwaitAwaiter...` general-purpose methods defined above, the following specialty await methods are also defined.
+
+```C#
+  namespace System.Runtime.CompilerServices
+  {
+      public static class RuntimeHelpers
```

Comment:
What assembly will define these? Should we restrict it to just system.private.corelib?

Reply:
In practice, this class already exists and is in corelib, so we will put these in corelib. As far as ECMA is concerned, I see no reason to mandate it.

Reply:
If there's no reason to mandate it, that must mean that the runtime needs to accept a user defining these helpers themselves, rather than the ones the runtime defines. That seems... Unlikely to be what you want.